File: Data Structures & Algorithms/find-peak-element/submission-2.java
class Solution {
    public int findPeakElement(int[] arr) {
        int n = arr.length;
        int l= 0;
        int r = n-1;
        while(l<r){
            int mid = l+(r-l)/2;
            if(arr[mid]<arr[mid+1]){
                l = mid+1;
            }
            else{
                r = mid;
            }
        }
        return l;
    }
}

File: Data Structures & Algorithms/remove-element/submission-2.py
class Solution:
    def removeElement(self, arr: List[int], val: int) -> int:
        k = 0
        i=0
        while(i<len(arr)):
            if arr[i]==val:
                i+=1
                continue
            else:
                arr[k] = arr[i]
                k+=1
                i+=1
            
        return k
